third_party/move/move-compiler-v2/src/pipeline/livevar_analysis_processor.rs

```diff
@@ -52,6 +52,18 @@ pub struct LiveVarInfoAtCodeOffset {
     pub after: BTreeMap<TempIndex, LiveVarInfo>,
 }
 
+impl LiveVarInfoAtCodeOffset {
+    /// Creates a set of the temporaries alive before this program point.
+    pub fn before_set(&self) -> BTreeSet<TempIndex> {
+        self.before.keys().cloned().collect()
```

Comment:
Should you check whether the bound value is a non-empty set here?
```
self.before.entries().filter_map(|(k, v) if (!v.usages.is_empty()) { Some(k) } else { None }).cloned().collect()
```
Or is it guaranteed that the set is non-empty if the entry exists?  If the latter is the case, please document it in LiveVarInfo below.

Reply:
Why would we need a guarantee for non-emptiness? If the set is empty its empty, which is a valid scenario. (There are program points indeed where no variable is in this set.)

Reply:
Ah I see what you mean. Yes, it is guaranteed that this set is not empty.

Reply:
Done, documented.